api/types/events/events.go

```diff
@@ -125,3 +125,30 @@ func (e *Exec) TrimToMaxSize(maxSize int) AuditEvent {
 
 	return out
 }
+
+// TrimToMaxSize trims the UserLogin event to the given maximum size.
+// The initial implementation is to cover concerns that a malicious user could
+// craft a request that creates error messages too large to be handled by the
+// underlying storage and thus cause the events to be omitted entirely. See
+// teleport-private#172.
+func (e *UserLogin) TrimToMaxSize(maxSize int) AuditEvent {
+	size := e.Size()
+	if size <= maxSize {
+		return e
+	}
+
+	out := proto.Clone(e).(*UserLogin)
+	out.Status.Error = ""
+	out.Status.UserMessage = ""
+
+	// Use 10% max size ballast + message size without Error and UserMessage
```

Comment:
I'm more interested in the logic behind the ballast than the actual number, could we briefly say why we need the ballast, and then why 10%?

Reply:
This is following the same pattern as the others. Rather than making this one special I'd like to follow up this PR with a minor refactoring PR that moves this out into a function with comments. I can also add ellipsis and return concrete types in the other TrimToMaxSize methods.

Reply:
Sounds good to me.